data/pools/transactionPool.go

```diff
@@ -351,79 +265,31 @@ func (pool *TransactionPool) OnNewBlock(block bookkeeping.Block) {
 	defer pool.mu.Unlock()
 
 	var stats telemetryspec.ProcessBlockMetrics
+	var knownCommitted uint
+	var unknownCommitted uint
 
-	remove := make(map[transactions.Txid]error, 0)
-
-	// collect transactions that appear in the last block
 	payset, err := block.DecodePayset()
-	if err != nil {
-		return
-	}
-
-	for _, tx := range payset {
-		txid := tx.ID()
-		_, ok := pool.pendingTxns[txid]
-		if ok {
-			remove[txid] = nil
-			stats.KnownCommittedCount++
-		} else {
-			stats.UnknownCommittedCount++
-		}
-	}
-
-	// collect expired transactions
-	expired := 0
-	for txid, tx := range pool.pendingTxns {
-		err := tx.Txn.Alive(block)
-		if err != nil {
-			remove[txid] = err
-			expired++
-			stats.ExpiredCount++
+	if err == nil {
+		for _, tx := range payset {
+			txid := tx.ID()
+			_, ok := pool.pendingTxns[txid]
+			if ok {
+				knownCommitted++
+			} else {
+				unknownCommitted++
+			}
 		}
 	}
 
-	// remove all collected transactions
-	for txid, txErr := range remove {
-		pool.remove(txid, txErr)
+	if pool.pendingBlockEvaluator == nil || block.Round() >= pool.pendingBlockEvaluator.Round() {
```

Comment:
Is this the correct Round() value?  There seems to be a potential for confusion between `pool.pendingBlockEvalutaor.Round()` and `pool.pendingBlockEvalutaorRound`.  It's not clear what the distinction is supposed to be between the two.

Reply:
Yeah, the terminology here isn't ideal.  I'll think about how to clean it up.

`pendingBlockEvaluator.Round()` is the round in which we started "testing" the transactions in the pool (i.e., tentatively adding them to a block to see if they can commit).  `pendingBlockEvaluatorRound` is the round in which a particular transaction might eventually get committed, if we have more than a block's worth of transactions in the pool.  The reason for tracking the latter value is so that we can check that those future transactions are still going to be valid in those later rounds (by their LastValid, in particular).